qt: Reset toolbar after all wallets are closed

hebasto opened the pull request:
If the last open wallet is closed from the non-"Overview" tab, that tab remains active when a new wallet is opened:

![Screenshot from 2020-05-06 09-00-26](https://user-images.githubusercontent.com/32963518/81142394-46821880-8f78-11ea-84b5-1d02f2b7f3f1.png)

This PR fixes this bug.

promag commented on src/qt/walletframe.cpp:
```diff
@@ -94,6 +94,10 @@ void WalletFrame::removeWallet(WalletModel* wallet_model)
     delete walletView;
+
+    if (mapWalletViews.empty()) {
+        gui->gotoOverviewPage();
```
I'd avoid using `gui`, it should be unknown to `WalletFrame` - actually I have a branch to remove `gui` here. May I suggest to do this in `BitcoinGUI::removeWallet`, in `m_wallet_selector->count() == 0` condition.

hebasto replied:
[Updated](https://github.com/bitcoin/bitcoin/pull/18896#issuecomment-625035798).

promag reviewed (commented):
~~No strong opinion.~~
Concept ACK, it fixes the bug when a wallet is opened after closing the last one where the current page isn't overview.

hebasto committed:
qt: Reset toolbar after all wallets are closed

hebasto commented:
Updated 7251b993093730f965f9de5b1c45fe3238e2dfd9 -> 1e9bfd4926a3cbb9db877c773a60b0bbbbe8bde0 ([pr18896.01](https://github.com/hebasto/bitcoin/commits/pr18896.01) -> [pr18896.02](https://github.com/hebasto/bitcoin/commits/pr18896.02), [diff](https://github.com/hebasto/bitcoin/compare/pr18896.01..pr18896.02)):

- addressed @promag's [comment](https://github.com/bitcoin/bitcoin/pull/18896#discussion_r421131508)

jonasschnelli commented:
utACK 1e9bfd4926a3cbb9db877c773a60b0bbbbe8bde0

hebasto commented:
Reopened to update Travis CI build list.

luke-jr commented:
utACK 1e9bfd4926a3cbb9db877c773a60b0bbbbe8bde0

hebasto commented:
@promag Mind re-reviewing this PR?

promag reviewed (commented):
Code review ACK 1e9bfd4926a3cbb9db877c773a60b0bbbbe8bde0.
